File: star1/bar_graph1.c
#include <stdio.h>  // 標準入出力関数

int main(int argc, char const *argv[]) {
  // 横棒グラフを表示します。
  // 直接、表示することも出来ます。
  printf("■■■■■\n");
  printf("■■■■□\n");
  printf("■■■□□\n");
  printf("■■□□□\n");
  printf("■□□□□\n");

  // ５つのグラフではなく７つのグラフにしたいなど、
  // 数が変わった時にも対応できるよう、
  // 繰り返し構文を使います。
  // 変数名は、i でも良いですが、
  // 行の変数であることが分かるようにすると良いです。
  int row;           // 行
  int black_number;  // 黒い■の数
  int white_number;  // 白い□の数

  for (row = 0; row < 5; row++) {
    black_number = 5 - row; // 黒い■の数
    white_number = row;     // 白い□の数
    // 初期値は設定済みなので、省略出来ます。
    for (; black_number > 0; black_number--) {
      printf("■");         // ■を表示します。
    }
    for (; white_number > 0; white_number--) {
      printf("□");         // ■を表示します。
    }
    printf("\n");           // 改行します。
  }

  return 0;
}

File: star1/bar_graph1.rb
# 横棒グラフを表示します。
# 直接、表示することも出来ます。
puts "■■■■■"
puts "■■■■□"
puts "■■■□□"
puts "■■□□□"
puts "■□□□□"

# Ruby では、for 文よりは、
# 範囲オブジェクトとeachメソッドの組み合わせ等が良く用いられます。
# (0...5).each で、行の変数rowには、0, 1, 2, 3, 4 が順に入ります。(5は含まない)
# (0..5).each  で、行の変数rowには、0, 1, 2, 3, 4, 5 が順に入ります。(5を含む)
(0...5).each do |row|
  black_number = 5 - row # 黒い■の数
  white_number = row     # 白い□の数

  # Rubyは文字列操作にも柔軟性があります。
  # 例えば "■" * 3 で "■■■" になります。
  # 文字列を連結する際には、「+」演算子を用いることも出来ます。
  print "■" * black_number + "□" * white_number + "\n"
end
